pwiz_tools/Skyline/TestFunctional/PastePeptidesWithChargesTest.cs

```diff
@@ -0,0 +1,105 @@
+                Tuple.Create("FIQPNIGELPTALK", 2, "sp|A2AGT5|CKAP5_MOUSE"),
+                Tuple.Create("LDDIFEPVLIPEPK", 2, "sp|A2AGT5|CKAP5_MOUSE"),
+                Tuple.Create("FYTQQWEDYR", 2, "sp|Q9WTX6|CUL1_MOUSE"),
+                Tuple.Create("ESFESQFLADTER", 2, "sp|Q9WTX6|CUL1_MOUSE"),
+                Tuple.Create("SPLAELGVLK", 2, "sp|Q8BHK9|ERC6L_MOUSE"),
+                Tuple.Create("ASLGPNLDLQDSVVLYHR", 3, "sp|Q8BHK9|ERC6L_MOUSE"),
+                Tuple.Create("AQGLDLLQAVLTR", 2, "sp|P60330|ESPL1_MOUSE"),
+                Tuple.Create("EVAQAATPQNPVNQGK", 2, "sp|Q8R080|GTSE1_MOUSE"),
+                Tuple.Create("GSQSDVLQDKPSTAPDAASR", 3, "sp|Q8R080|GTSE1_MOUSE"),
+                Tuple.Create("LAASASTQQFQEVK", 2, "sp|P97329|KI20A_MOUSE"),
+                Tuple.Create("TPTC[+57]QSSTDSSPYAR", 2, "sp|P97329|KI20A_MOUSE"),
+                Tuple.Create("LNLVDLAGSENIGR", 2, "sp|Q6P9P6|KIF11_MOUSE"),
+                Tuple.Create("EAGNINQSLLTLGR", 2, "sp|Q6P9P6|KIF11_MOUSE"),
+                Tuple.Create("LTDDHVQFLIYQILR", 3, "sp|P47811|MK14_MOUSE"),
+                Tuple.Create("SLAPVELLLR", 2, "sp|P70268|PKN1_MOUSE"),
+                Tuple.Create("KLEEEEEGSAPATSR", 2, "sp|A2APB8|TPX2_MOUSE"),
+                Tuple.Create("NVTQAEPFSLETDK", 2, "sp|A2APB8|TPX2_MOUSE"),
+                Tuple.Create("EVTTLTADPPDGIK", 2, "sp|Q921J4|UBE2S_MOUSE"),
+                Tuple.Create("LLTEIHGGAC[+57]STSSGR", 3, "sp|Q921J4|UBE2S_MOUSE"),
+                Tuple.Create("LAEQFAAGEVLTDMSR", 2, "sp|Q80X41|VRK1_MOUSE"),
+                Tuple.Create("SVESQGAIHGSMSQPAAGC[+57]SSSDSSR", 3, "sp|Q80X41|VRK1_MOUSE"),
+            };
+            var pasteDlg = ShowDialog<PasteDlg>(SkylineWindow.ShowPastePeptidesDlg);
+            var clipboardText = TextUtil.LineSeparate(peptideChargeProteins.Select(tuple =>
+                tuple.Item1 + Transition.GetChargeIndicator(tuple.Item2) + "\t" + tuple.Item3
+            ));
+            RunUI(() =>
+            {
+                SetClipboardText(clipboardText);
+                pasteDlg.PastePeptides();
+            });
+            OkDialog(pasteDlg, pasteDlg.OkDialog);
+            Assert.AreEqual(SkylineWindow.Document.MoleculeCount, peptideChargeProteins.Length);
+
+            // Make sure that the peptides were all inserted with the specified charges
+            foreach (var tuple in peptideChargeProteins)
+            {
+                var unmodifiedSequence = FastaSequence.StripModifications(tuple.Item1);
+                var peptideDocNodes =
+                    SkylineWindow.Document.Molecules.Where(pep => pep.Peptide.Sequence == unmodifiedSequence).ToList();
+                Assert.AreEqual(1, peptideDocNodes.Count);
+                var peptideDocNode = peptideDocNodes[0];
+                Assert.AreEqual(1, peptideDocNode.TransitionGroupCount);
+                var precursor = peptideDocNode.TransitionGroups.First();
+                Assert.AreEqual(tuple.Item2, precursor.PrecursorCharge);
```

Comment:
Add a check that charge 2 precursors are auto-select (since that is the default setting), and non-charge 2 precursors are not auto-select. You might also repeat the test with the document settings changed to only charge 3 precursors and check that produces the right auto-select settings, and also with multiple charges (e.g. "2, 3, 4") and check that all resulting peptides are not auto-select.